ckan/plugins/toolkit.py

```diff
@@ -220,17 +212,19 @@ def check_ckan_version(
     :type max_version: string
 
     """
-    current = _version_str_2_list(ckan.__version__)
-
-    if min_version:
-        min_required = _version_str_2_list(min_version)
-        if current < min_required:
-            return False
-    if max_version:
-        max_required = _version_str_2_list(max_version)
-        if current > max_required:
-            return False
-    return True
+    current = pv.parse(ckan.__version__)
+    try:
+        at_least_min = (min_version is None) or (current >= pv.parse(min_version))
+    except pv.InvalidVersion:
+        # TODO: This should be a custom error per the CKAN dev guidelines
```

Comment:
Please, finalize the PR. Instead of adding new TODO's into codebase, implement whatever you suggesting here